src/libraries/Common/src/System/Security/Cryptography/Asn1Reader/AsnReader.Integer.cs

```diff
@@ -564,7 +565,7 @@ public bool TryReadUInt8(Asn1Tag expectedTag, out byte value)
             }
 
             // Slice first so that an out of bounds value triggers a CryptographicException.
-            ReadOnlyMemory<byte> contents = Slice(_data, headerLength, length.Value);
+            ReadOnlyMemory<byte> contents = Slice(_data, headerLength, length!.Value);
```

Comment:
Why did you choose to use `length!.Value` here, but `Debug.Assert(length.HasValue)` in a simlar case above?

Reply:
I thought one of the if check conditions above `length < 1` would make sure it would have value

Reply:
I see.  Yes.  Thanks.